### PR TITLE
Feature/gh 155 refactoring

### PR DESCRIPTION
**Pull request by marartz**

### CODE REVIEW

File: src/Gastromio.App/ClientApp/src/app/restaurant-admin/restaurant-admin.facade.ts [hunk 1]
```diff
@@ -103,22 +102,16 @@ export class RestaurantAdminFacade {
             return 0;
           });
 
+          console.log("restaurant: ", restaurant);
```
Comment (ikke09) [inline src/Gastromio.App/ClientApp/src/app/restaurant-admin/restaurant-admin.facade.ts:105]: Missed debug log? 🧹

**Review by ikke09 — CHANGES_REQUESTED** [review]
Great Work @marartz 💪🏽🚀
Pull request is a little bit to big to cover every detail in the review. Hopefully I coverd the most crucial aspects here! I made some smaller comments. Please take a look at them 😉

File: src/Gastromio.Core/Domain/Model/Orders/OrderId.cs [hunk 2]
```diff
@@ -7,6 +7,8 @@ public class OrderId : ValueType<Guid>
     {
         public OrderId(Guid value) : base(value)
         {
+            if (value == Guid.Empty)
+                throw new InvalidOperationException("order id is invalid");
```
[the remark this answers is not in the document]
Comment (marartz) [reply src/Gastromio.Core/Domain/Model/Orders/OrderId.cs:11]: You are right. I changed it accordingly.

File: src/Gastromio.Core/Domain/Model/Restaurants/Restaurant.cs [hunk 3]
```diff
@@ -425,250 +269,189 @@ public bool IsOrderPossibleAt(DateTimeOffset orderDateTime)
...
-
-            if (pickupInfo.MinimumOrderValue.HasValue && pickupInfo.MinimumOrderValue < 0)
-                return FailureResult<bool>.Create(FailureResultCode.RestaurantMinimumPickupOrderValueTooLow);
-            if (pickupInfo.MinimumOrderValue.HasValue && pickupInfo.MinimumOrderValue > 50)
-                return FailureResult<bool>.Create(FailureResultCode.RestaurantMinimumPickupOrderValueTooHigh);
-
-            if (pickupInfo.MaximumOrderValue.HasValue && pickupInfo.MaximumOrderValue < 0)
-                return FailureResult<bool>.Create(FailureResultCode.RestaurantMaximumPickupOrderValueTooLow);
-
             PickupInfo = pickupInfo;
```
[the remark this answers is not in the document]
Comment (marartz) [reply src/Gastromio.Core/Domain/Model/Restaurants/Restaurant.cs:274]: I opted for a domain exception here.

File: src/Gastromio.Core/Domain/Model/Restaurants/Restaurant.cs [hunk 4]
```diff
@@ -155,235 +137,97 @@ public IReadOnlyCollection<ExternalMenu> ExternalMenus
...
-
-            if (string.IsNullOrEmpty(contactInfo.EmailAddress))
-                return FailureResult<bool>.Create(FailureResultCode.RestaurantEmailRequired);
-            if (!Validators.IsValidEmailAddress(contactInfo.EmailAddress))
-                return FailureResult<bool>.Create(FailureResultCode.RestaurantEmailInvalid, contactInfo.EmailAddress);
-
-            if (!string.IsNullOrEmpty(contactInfo.Mobile) && !Validators.IsValidPhoneNumber(contactInfo.Mobile))
-                return FailureResult<bool>.Create(FailureResultCode.RestaurantMobileInvalid, contactInfo.Mobile);
-
             ContactInfo = contactInfo;
```
[the remark this answers is not in the document]
Comment (marartz) [reply src/Gastromio.Core/Domain/Model/Restaurants/Restaurant.cs:157]: done

File: src/Gastromio.Core/Domain/Model/Users/User.cs [hunk 5]
```diff
@@ -62,30 +65,30 @@ UserId updatedBy
...
+            return validationResult;
         }
 
-        public Result<bool> ChangePassword(string password, bool checkPasswordPolicy, UserId changedBy)
+        public void ChangePassword(string password, bool checkPasswordPolicy, UserId changedBy)
         {
             if (password == null)
                 throw new ArgumentNullException(nameof(password));
 
             if (checkPasswordPolicy && !passwordPolicyRegex.IsMatch(password))
```
[the remark this answers is not in the document]
Comment (marartz) [reply src/Gastromio.Core/Domain/Model/Users/User.cs]: Done

File: src/Gastromio.Core/Domain/Model/Users/User.cs [hunk 6]
```diff
@@ -62,30 +65,30 @@ UserId updatedBy
...
             UpdatedBy = changedBy;
-
-            return SuccessResult<bool>.Create(true);
         }
 
-        public Result<bool> ValidatePassword(string password)
+        public bool ValidatePassword(string password)
         {
-            byte[] testPasswordHash = PBKDF2(password, PasswordSalt, PBKDF2_ITERATIONS, HASH_BYTES);
+            var testPasswordHash = PBKDF2(password, PasswordSalt, PBKDF2_ITERATIONS, HASH_BYTES);
```
[the remark this answers is not in the document]
Comment (marartz) [reply src/Gastromio.Core/Domain/Model/Users/User.cs:77]: Ok... We'll do it in the next refactoring.

File: src/Gastromio.App/ClientApp/src/app/restaurant-admin/restaurant-admin.facade.ts (hunk 1 above)
Comment (marartz) [reply src/Gastromio.App/ClientApp/src/app/restaurant-admin/restaurant-admin.facade.ts]: Done

**Comment by marartz** [comment]
@ikke09 : I'm done with your requested changes and comments. Could you review again?

**Review by ikke09 — APPROVED** [review]
Seems all good now! 🚀

**Pull request merged**